manifests/database.pp

```diff
@@ -257,7 +257,6 @@
           group   => $group,
           before  => Exec["oracle database ${title}"],
         }
-      }
```

Comment:
I don't know about this 
$template_variables is optional and now you need to use it else you cannot use dbt. you can still have a dbt template with no variables.

Reply:
can you what the real issue because all template command have -responsefile NO_VALUE only non template go the starter db's section

Reply:
The real issue is when I try to create DB Instance with template, but without template variables, I get error:

'su - oracle -c "/bin/ksh -c \"/local/oracle/product/12.2.0.1/dbse_1/bin/dbca -silent -createDatabase -templateName /local/install/dbtemplate_12.2_orc12.dbt -gdbname orc12 -sid orc12 -characterSet AL32UTF8 -responseFile NO_VALUE -sysPassword password -systemPassword password -dbsnmpPassword Welcome01 -asmsnmpPassword Welcome01 -storageType FS -emConfiguration NONE /local/oracle/oradata -initParams job_queue_processes=4,open_cursors=1000,processes=600 \""' returned 254 instead of one of [0]

However it still generates response file. 
Yes, it seems that with my fix templates becomes mandatory, but in my case this command worked
$command = "${elevation_prefix}${oracle_home}/bin/dbca -silent -createDatabase -responseFile ${download_dir}/database_${sanitized_title}.rsp${elevation_suffix}"

Reply:
I think you are hitting the init params bug , where they are not used or ignored that why I introduced the param file that somehow always works.

but what is this /local/oracle/oradata in your command
 -emConfiguration NONE /local/oracle/oradata -initParams job_queu 

I see the bug in my code with the optional data_file_destination 
$command = "${command_pre} ${data_file_destination} ${command_var} ${command_init} ${command_nodes} ${elevation_suffix}"

this should be 

$command = "${command_pre} ${command_data_file} ${command_var} ${command_init} ${command_nodes} ${elevation_suffix}"

Reply:
So let's make it optional again, can you also change the above and show me the real error of the dbca statement.

Reply:
Seems that change above fixed the issue.

Reply:
nice can you do the changes so we can merge this. 
thanks for finding this